ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCaseBuilderUtil.java

```diff
@@ -197,7 +200,11 @@ private static Topic createTopicFromStatement(
 
         if (isCsOrCT(stmt)) {
           final PreparedStatement<?> prepare = parser.prepare(stmt, metaStore);
-          topics.add(extractTopic.apply(prepare));
+          final ConfiguredStatement<?> configured =
+              ConfiguredStatement.of(prepare, Collections.emptyMap(), ksqlConfig);
+          // TODO: cleaner way to handle this?
+          final ConfiguredStatement<?> withFormats = new DefaultFormatInjector().inject(configured);
```

Comment:
This is needed or else tests that don't explicitly specify `KEY_FORMAT` in CS/CT statements fail. Feels hacky to add the injector here but I can't think of a better solution (besides requiring explicit `KEY_FORMAT` in all tests, which doesn't seem right).

Reply:
Welcome to the mess of the testing tool! 

You're approach is generally correct - we need your injector in QTT to do its thing, so unfortunately, this means adding it to QTT, given QTT doesn't currently just use the default `Injectors` class, (not sure it actually can... not something I'm that familiar with - @agavra  may have more context).

For background: this bit of code is attempting to extract the topic info from the statement.  Later in `TestExecutorUtil` we run the statement for real.  Ideally, we don't want to have to add your injector to both places.  That's just making the code worse.

Maybe an OK approach would be to have a common method (probably in `TestExecutorUtil`), that knows how to prepare the statement, with the correct injectors.  You add your injector there, and have both places call that method.

Not ideal, but....